misc.go

```diff
@@ -47,7 +50,7 @@ func runeSliceWidthRange(rs []rune, beg, end int) []rune {
 		}
 		curr += w
 	}
-	return nil
+	return rs[b:curr]
```

Comment:
This doesn't look right - `curr` is a *cell* index (unlike `b` which is a *rune* index) so it doesn't make sense to use `curr` here for indexing `rs`. I agree the current implementation doesn't work for a few edge cases, and have suggested an alternate implementation in #1270, but I don't know if it will get accepted or not.